src/Manager/DocumentManager.php

```diff
@@ -1463,8 +1469,9 @@ public function getTransformValue($source, $ruleField)
                 $targetId = $this->getTargetId($ruleField, $source[$ruleField['field_name_source']]);
                 if (!empty($targetId['record_id'])) {
                     return $targetId['record_id'];
-                } else {
```

Comment:
**issue**:  Removing the **else** statement altogether means we're no longer catching other potential options, which I think could be dangerous 
**suggestion**: Add back the **else** statement below the **elseif** statement in order to code defensively with a more generic Exception thrown. 
**suggestion**:  Add code to log error messages with $this->logger->error(.....)

Reply:
Thanks @EstelleMyddleware. 
Could you change the codelike this : 
`$targetId = $this->getTargetId($ruleField, $source[$ruleField['field_name_source']]);
                if (!empty($targetId['record_id'])) {
                    return $targetId['record_id'];
                // No need of relate field in case of deletion
				} elseif ($this->documentType != 'D') {
                     throw new \Exception('Target id not found for id source '.$source[$ruleField['field_name_source']].' of the rule '.$ruleField['field_id']);
                				} else {
                     return null;
                }
`
Thanks.